### PR TITLE
Update web.config, add publish tool, convert to portable

### PR DESCRIPTION
**Pull request by BrennanConroy**
@JunTaoLuo @Tratcher @pakrym @Eilon

### CODE REVIEW

File: samples/SignalRSample.Web/project.json [hunk 1]
```diff
@@ -14,18 +14,30 @@
   },
   "frameworks": {
     "net451": {},
-    "netstandardapp1.5": {
+    "netcoreapp1.0": {
       "imports": [
         "dnxcore50"
```
Comment (pakrym) [inline samples/SignalRSample.Web/project.json]: Why?

File: test/Microsoft.AspNetCore.SignalR.Messaging.Tests/project.json [hunk 2]
```diff
@@ -10,13 +10,17 @@
     "keyFile": "../../tools/Key.snk"
   },
   "frameworks": {
-    "netstandardapp1.5": {
+    "netcoreapp1.0": {
       "imports": [
         "dnxcore50",
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.SignalR.Messaging.Tests/project.json]: Same as above

File: test/Microsoft.AspNetCore.SignalR.Server.Tests/project.json [hunk 3]
```diff
@@ -12,13 +12,17 @@
...
         "dnxcore50",
         "portable-net451+win8"
       ],
       "dependencies": {
         "dotnet-test-xunit": "1.0.0-*",
+        "Microsoft.NETCore.App": {
+          "version": "1.0.0-*",
+          "type": "platform"
+        },
         "moq.netcore": "4.4.0-beta8"
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.SignalR.Server.Tests/project.json]: Usually you need a dependency on `System.Diagnostics.TraceSource` for `moq.netcore` to work on coreclr.

File: test/Microsoft.AspNetCore.SignalR.Messaging.Tests/project.json (hunk 2 above)
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.SignalR.Messaging.Tests/project.json]: This one is "normal" https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.Test/project.json#L17

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by BrennanConroy** [comment]
🆙 📅

File: test/Microsoft.AspNetCore.SignalR.Server.Tests/project.json [hunk 4]
```diff
@@ -12,14 +12,19 @@
...
-        "moq.netcore": "4.4.0-beta8"
+        "Microsoft.NETCore.App": {
+          "version": "1.0.0-*",
+          "type": "platform"
+        },
+        "moq.netcore": "4.4.0-beta8",
+        "System.Diagnostics.TraceSource": "4.0.0-*"
       }
     },
     "net451": {
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.SignalR.Server.Tests/project.json]: `"Microsoft.NETCore.Platforms": "1.0.1-*"` ?

**Comment by pakrym** [comment]
:shipit:

**Pull request merged**